src/common/AndroidUtils.ts

```diff
@@ -154,157 +139,147 @@ export class AndroidSDKUtils {
+                const matchingPlatforms = packages.platforms.filter((pkg) =>
+                    pkg.version.sameOrNewer(minSupportedRuntimeAndroid)
                 );
-            }
-
-            // Sort the packages with latest version by negating the comparison result
-            matchingPlatforms.sort((a, b) => a.version.compare(b.version) * -1);
+                if (matchingPlatforms.length < 1) {
+                    return Promise.reject(
+                        new Error(
+                            `Could not locate a supported Android API package. Minimum supported Android API package version is ${androidConfig.minSupportedRuntimeAndroid}`
+                        )
+                    );
+                }
 
-            // Return the latest package that also has matching emulator images
-            for (const platform of matchingPlatforms) {
-                const emulatorImage = await AndroidSDKUtils.packageWithRequiredEmulatorImages(
-                    platform
+                // Sort the packages with latest version by negating the comparison result
+                matchingPlatforms.sort(
+                    (a, b) => a.version.compare(b.version) * -1
                 );
 
-                if (emulatorImage) {
-                    return platform;
+                // Return the latest package that also has matching emulator images
+                for (const platform of matchingPlatforms) {
+                    const emulatorImage = await AndroidSDKUtils.packageWithRequiredEmulatorImages(
+                        platform
+                    );
+
+                    if (emulatorImage) {
+                        return Promise.resolve(platform);
+                    }
                 }
-            }
 
-            // If we got here then it means that we don't have any Android API packages with emulator images.
-            // So we will go ahead and return the latest one anyway. This is b/c the setup command will error
-            // out stating that no packages with matching emulator images have been found so the user would
-            // then know that they should install emulator images for the latest API package.
-            return matchingPlatforms[0];
-        } catch (error) {
-            return Promise.reject(
-                new Error(
-                    `Could not find android api packages. ${error.errorMessage}`
+                // If we got here then it means that we don't have any Android API packages with emulator images.
+                // So we will go ahead and return the latest one anyway. This is b/c the setup command will error
+                // out stating that no packages with matching emulator images have been found so the user would
+                // then know that they should install emulator images for the latest API package.
+                return Promise.resolve(matchingPlatforms[0]);
+            })
+            .catch((error) =>
+                Promise.reject(
```

Comment:
Why would you not `return` the Promise here?

Reply:
Hmm... looks like I need to work on the logic here a bit more. Previously we had a try-catch block b/c after going through all of the checks (which returned Promise.reject) we could still call into `AndroidSDKUtils.packageWithRequiredEmulatorImages()` which can throw instead of coming back with no results so for that scenario we were catching the thrown error and returning it here. But as you pointed out, I think we can use some more cleanup here.